pkg/sql/types/types.go

```diff
@@ -1630,6 +1630,7 @@ func (t *T) upgradeType() error {
 		if t.InternalType.Precision > 0 {
 			t.InternalType.TimePrecisionIsSet = true
 		}
+		t.InternalType.Oid = familyToOid[t.Family()]
 	case IntervalFamily:
```

Comment:
We also need this same thing for Intervals.

Reply:
Actually, I have a better idea. Can you instead remove the `default` case, and move it into an block after the switch statement, conditional on the Oid still being 0? That would protect us from future problems like this.

Reply:
should be able to add tests to this as well (there are upgradeType tests in types_test.go)

Reply:
I added tests that test an empty type with just the family always gets the correct Oid set - this was failing on timestamps, intervals and arrays before. I also added a test that the Oid for a BitFamily type with visible varbit was not set to the default family ID, but was set to the correct one.